src/yamlfix/adapters.py

```diff
@@ -379,12 +381,27 @@ def _ruamel_yaml_fixer(self, source_code: str) -> str:
         # Return the output to a string
         string_stream = StringIO()
         for source_dict in source_dicts:
+            self._quote_gha_container_volumes(source_dict)
             self.yaml.dump(source_dict, string_stream)
             source_code = string_stream.getvalue()
         string_stream.close()
 
         return source_code.strip()
 
+    def _quote_gha_container_volumes(
+        self, source_dict: CommentedMap | CommentedSeq
+    ) -> None:
+        """Quote jobs.*.container.volumes entries."""
```

Comment:
I find the phrasing of the docstring confusing. Wouldn't it be clearer something like `prevent list strings to be converted to dictionaries`?

I'd add an example of what this function fixes (the one in the issue is just fine)

I'd also change the name of the method, as it's not specific to gha_container_volumes, there may be other strings with colons that will benefit from this method